sbpy/utils/dastcom5.py

```diff
@@ -0,0 +1,575 @@
+    column2 = (record, a, ecc, inc, raan, argp, m, epoch)
+    column1 = ("record", "a", "ecc", "inc", "raan", "argp", "m", "EPOCH")
+    data = Table(rows=[column1, column2])
+
+    return data
+
+
+def record_from_name(name):
+    """Search `dastcom.idx` and return logical records that
+    match a given string.
+
+    Body name, SPK-ID, or alternative designations can be used.
+
+    Parameters
+    ----------
+    name : str
+        Body name.
+
+    Returns
+    -------
+    records : list (int)
+        DASTCOM5 database logical records matching str.
+
+    """
+    records = []
+    lines = string_record_from_name(name)
+    for line in lines:
+        records.append(int(line[:6].lstrip()))
+    return records
+
+
+def string_record_from_name(name):
+    """Search `dastcom.idx` and return body full record.
+
+    Search DASTCOM5 index and return body records that match string,
+    containing logical record, name, alternative designations, SPK-ID, etc.
+
+    Parameters
+    ----------
+    name : str
+        Body name.
+
+    Returns
+    -------
+    lines: list(str)
+        Body records
+    """
+
+    idx_path = os.path.join(DBS_LOCAL_PATH, "dastcom.idx")
+    lines = []
+    with open(idx_path, "r") as inF:
+        for line in inF:
+            if re.search(r"\b" + name.casefold() + r"\b", line.casefold()):
+                lines.append(line)
+    return lines
+
+
+def read_headers():
+    """Read `DASTCOM5` headers and return asteroid and comet headers.
+
+    Headers are two numpy arrays with custom dtype.
+
+    Returns
+    -------
+    ast_header, com_header : tuple (numpy.ndarray)
+        DASTCOM5 headers.
+
+    """
+
+    ast_path = os.path.join(DBS_LOCAL_PATH, "dast5_le.dat")
+    ast_dtype = np.dtype(
+        [
+            ("IBIAS1", np.int32),
+            ("BEGINP1", "|S8"),
+            ("BEGINP2", "|S8"),
+            ("BEGINP3", "|S8"),
+            ("ENDPT1", "|S8"),
+            ("ENDPT2", "|S8"),
+            ("ENDPT3", "|S8"),
+            ("CALDATE", "|S19"),
+            ("JDDATE", np.float64),
+            ("FTYP", "|S1"),
+            ("BYTE2A", np.int16),
+            ("IBIAS0", np.int32),
+        ]
+    )
+
+    with open(ast_path, "rb") as f:
+        ast_header = np.fromfile(f, dtype=ast_dtype, count=1)
+
+    com_path = os.path.join(DBS_LOCAL_PATH, "dcom5_le.dat")
+    com_dtype = np.dtype(
+        [
+            ("IBIAS2", np.int32),
+            ("BEGINP1", "|S8"),
+            ("BEGINP2", "|S8"),
+            ("BEGINP3", "|S8"),
+            ("ENDPT1", "|S8"),
+            ("ENDPT2", "|S8"),
+            ("ENDPT3", "|S8"),
+            ("CALDATE", "|S19"),
+            ("JDDATE", np.float64),
+            ("FTYP", "|S1"),
+            ("BYTE2C", np.int16),
+        ]
+    )
+
+    with open(com_path, "rb") as f:
+        com_header = np.fromfile(f, dtype=com_dtype, count=1)
+
+    return ast_header, com_header
+
+
+def read_record(record):
```

Comment:
I couldn't get this function to work. It fails with 

>   File ".../sbpy/utils/dastcom5.py", line 502, in read_record
>     f.seek(phis_rec, os.SEEK_SET)

This function would be potentially useful and could output as `sbpy.data.Phys` object.

Reply:
Will work on it

Reply:
This function is already being used in orbit_from_record :sweat_smile:

Reply:
This error pops when the record is not there in the database. Please try with : 
```
read_record(1636)
```
which is for atira